docs/technologies.md

```diff
@@ -0,0 +1,423 @@
+        version='0.1.0',
+        packages=setuptools.find_namespace_packages(include=["your_project_name.*"])
+)
+```
+
+The generated setup is minimalistic. If you want to provide requirements for your Beam process, you can do it through the
+`SetupOptions`. You can store the requirements for your processes in the [`resources`](./project_setup.py#project-structure) directory.
+
+```python
+options.view_as(SetupOptions).requirements_file = resolve(get_resource_absolute_path('requirements.txt', Path(__file__)))
+```
+
+## BigQuery
+
+### Overview
+
+BigFlow provides vast support for BigQuery. Example use cases:
+
+* Ad-hoc queries (fits well in a Jupyter notebook)
+* Convenient BigQuery client 
+* Creating data processing pipelines
+* Creating BigQuery sensors
+
+The project starter generates the workflow called `internationalports`. It's a purely BigQuery based workflow.
+
+The workflow fits the single `internationalports.workflow` module:
+
+```python
+from bigflow import Workflow
+from bigflow.bigquery import DatasetConfig, Dataset
+
+dataset_config = DatasetConfig(
+    env='dev',
+    project_id='your-project-id',
+    dataset_name='internationalports',
+    internal_tables=['ports', 'polish_ports'],
+    external_tables={})
+
+dataset: Dataset = dataset_config.create_dataset_manager()
+
+create_polish_ports_table = dataset.create_table('''
+    CREATE TABLE IF NOT EXISTS polish_ports (
+      port_name STRING,
+      port_latitude FLOAT64,
+      port_longitude FLOAT64)
+''')
+
+create_ports_table = dataset.create_table('''
+    CREATE TABLE IF NOT EXISTS ports (
+      port_name STRING,
+      port_latitude FLOAT64,
+      port_longitude FLOAT64,
+      country STRING,
+      index_number STRING)
+''')
+
+select_polish_ports = dataset.write_truncate('ports', '''
+        SELECT port_name, port_latitude, port_longitude
+        FROM `{more_ports}`
+        WHERE country = 'POL'
+        ''', partitioned=False)
+
+populate_ports_table = dataset.collect('''
+        INSERT INTO `{more_ports}` (port_name, port_latitude, port_longitude, country, index_number)
+        VALUES 
+        ('GDYNIA', 54.533333, 18.55, 'POL', '28740'),
+        ('GDANSK', 54.35, 18.666667, 'POL', '28710'),
+        ('SANKT-PETERBURG', 59.933333, 30.3, 'RUS', '28370'),
+        ('TEXAS', 34.8, 31.3, 'USA', '28870');
+        ''')
+
+
+internationalports_workflow = Workflow(
+        workflow_id='internationalports',
+        definition=[
+                create_ports_table.to_job(id='create_ports_table'),
+                create_polish_ports_table.to_job(id='create_polish_ports_table'),
+                populate_ports_table.to_job(id='populate_ports_table'),
+                select_polish_ports.to_job(id='select_polish_ports'),
+        ],
+        schedule_interval='@once')
+```
+
+There are two notable elements in the `internationalports.workflow` module:
+
+* `DatasetConfig` class which defines a BigQuery dataset you want to interact with
```

Comment:
`DatasetConfig`  - link to  github source?

Reply:
done